ifttt/tests/test_api.py

```diff
@@ -24,16 +25,36 @@ def test_create_ifttt_enter(self):
 
         assert response.status_code == 200, 'Expect 200OK'
 
+        icr_entry = IncomingRequest.objects.get(user=4)
+
+        assert icr_entry is not None, 'Expect the Entry Record to Exist'
+
     def test_create_ifttt_exit(self):
+        #Setup
+        data ={"user": "4",
+               "project_id": "2",
+               "project_task_id": "23",
+               "time": "08:12:00 12-09-2015",
+               "entered_or_exited": "entered"}
+
+        IncomingRequest.objects.create(user=4, payload=json.dumps(data))
+
         c = Client()
-        response = c.post('/ifttt/', { 	"user": "4",
-                                          "project_id": "2",
-                                          "project_task_id": "23",
-                                          "time": "17:12:00 12-09-2015",
-                                          "entered_or_exited": "exited"})
+        exit_payload = {  "user": "4",
+                          "project_id": "2",
+                          "project_task_id": "23",
+                          "time": "17:12:00 12-09-2015",
+                          "entered_or_exited": "exited"
+                        }
+        response = c.post('/ifttt/', exit_payload)
 
         assert response.status_code == 200, 'Expect 200OK'
 
+        icr_entry = IncomingRequest.objects.filter(user=4)[1]
```

Comment:
`IncomingRequest.objects.get` requires that only once instance is returned. The original query you had would fail because it would return both the entry and exit items. 

I also added an assertion to make sure that the payload is what we saved.